### PR TITLE
20818 Fixed draft amalgamation initial values

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#20818

*Description of changes:*
- app version = 7.1.11
- created better getAccountId store getter and used it in place of old code
- don't save empty legalName in draft regular amalgamation
- don't save null correctNameOption in draft regular amalgamation
- improved email/phone logic

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-647-a791jt1b.web.app

File: src/components/Dashboard/TodoList.vue [hunk 1]
```diff
@@ -550,7 +550,7 @@
               v-if="EnumUtilities.isPayMethodOnlineBanking(item)"
               :filing="item"
               :payApiUrl="getPayApiUrl"
-              :accountId="accountId"
+              :accountId="getAccountId"
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:553]: Now using store getter instead of local code.

File: src/components/Dashboard/TodoList.vue [hunk 2]
```diff
@@ -996,9 +997,8 @@ export default class TodoList extends Mixins(AllowableActionsMixin, DateMixin, E
     }
 
     // load all the invitations here and push them into todo items
-    this.accountId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT'))?.id
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:999]: This isn't ideal code. The store getter is better code.

File: src/components/Dashboard/TodoList.vue [hunk 3]
```diff
@@ -996,9 +997,8 @@ export default class TodoList extends Mixins(AllowableActionsMixin, DateMixin, E
...
     const response =
-      await AuthServices.fetchAffiliationInvitations(this.getAuthApiUrl, this.getIdentifier, this.accountId)
+      await AuthServices.fetchAffiliationInvitations(this.getAuthApiUrl, this.getIdentifier, +this.getAccountId)
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:1001]: Using `+` to convert a string to a number on the fly.

File: src/components/Dashboard/TodoList/PaymentPendingOnlineBanking.vue [hunk 4]
```diff
@@ -43,7 +43,7 @@ export default class PaymentPendingOnlineBanking extends Vue {
   /** The subject filing. */
   @Prop({ required: true }) readonly filing!: any
   @Prop({ required: true }) readonly payApiUrl!: string
-  @Prop({ required: true }) readonly accountId!: any
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList/PaymentPendingOnlineBanking.vue:46]: The parent passes this in as a string, so I updated this declaration.

File: src/views/AmalgamationSelection.vue [hunk 5]
```diff
@@ -220,17 +222,25 @@ export default class AmalgamationSelection extends Vue {
...
    */
   private async createBusinessAA (type: AmalgamationTypes): Promise<string> {
-    const accountId = +JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT'))?.id || 0
-    const email = this.isShortFormAmalgamation(type) ? this.getBusinessEmail : ''
-    const phone = this.isShortFormAmalgamation(type) ? this.getFullPhoneNumber : ''
-    const legalName = this.isShortFormAmalgamation(type) ? this.getLegalName : ''
-    const correctNameOption = this.isShortFormAmalgamation(type) ? CorrectNameOptions.CORRECT_AML_ADOPT : null
+    // short-form amalgamations use the legal name of the primary business,
+    // otherwise don't set a legal name
+    const legalName = this.isShortFormAmalgamation(type) ? this.getLegalName : undefined
```
Comment (severinbeauvais) [inline src/views/AmalgamationSelection.vue:227]: Setting an object property to `undefined` is almost like removing it. Certainly, when the network code converts the object to JSON, the property is not saved.

File: src/views/AmalgamationSelection.vue [hunk 6]
```diff
@@ -220,17 +222,25 @@ export default class AmalgamationSelection extends Vue {
...
+
+    // short-form amalgamations adopt the name of the primary business,
+    // otherwise don't set a correct name option
+    const correctNameOption =
+      this.isShortFormAmalgamation(type) ? CorrectNameOptions.CORRECT_AML_ADOPT : undefined
+
+    // short-form amalgamations use the email and phone of the primary business (if they exist),
+    // otherwise set empty initial values
+    const email = (this.isShortFormAmalgamation(type) && this.getBusinessEmail) || ''
+    const phone = (this.isShortFormAmalgamation(type) && this.getFullPhoneNumber) || ''
```
Comment (severinbeauvais) [inline src/views/AmalgamationSelection.vue:237]: The old code assigned `''` here for non-short form amalgamation, and fell back to `''` below if getBusinessEmail or getFullPhoneNumber returned falsy.

I just combined it all here. Example:
| isShortFormAmalgamation | getBusinessEmail | result |
| - | - | - |
| true | '123' | '123' |
| true | '' | '' |
| false | '123' | '' |
| false | '' | '' |

File: src/views/AmalgamationSelection.vue [hunk 7]
```diff
@@ -220,17 +222,25 @@ export default class AmalgamationSelection extends Vue {
...
-    const phone = this.isShortFormAmalgamation(type) ? this.getFullPhoneNumber : ''
-    const legalName = this.isShortFormAmalgamation(type) ? this.getLegalName : ''
-    const correctNameOption = this.isShortFormAmalgamation(type) ? CorrectNameOptions.CORRECT_AML_ADOPT : null
+    // short-form amalgamations use the legal name of the primary business,
+    // otherwise don't set a legal name
+    const legalName = this.isShortFormAmalgamation(type) ? this.getLegalName : undefined
+
+    // short-form amalgamations adopt the name of the primary business,
+    // otherwise don't set a correct name option
+    const correctNameOption =
```
Comment (severinbeauvais) [inline src/views/AmalgamationSelection.vue:231]: There's also no point in saving correctNameOption=null if this isn't a short-form amalgamation.

**Comment by severinbeauvais** [comment]
Testing coverage is still failing, but all 3 examples I provided above work and the unit tests pass locally:

image

**Review by jamespaologarcia — APPROVED** [review]
LGTM

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**